gap/attributes/isorms.gi

```diff
@@ -663,119 +663,245 @@ end);
 # 4. Isomorphisms
 #############################################################################
 
-InstallMethod(IsomorphismSemigroups, "for Rees matrix semigroups",
-[IsReesMatrixSemigroup and IsWholeFamily,
- IsReesMatrixSemigroup and IsWholeFamily],
-function(R1, R2)
-  local mat, m, n, g, g1, g2, iso, isograph, isogroup, map, l, tup,
-  RMSInducedFunction;
-
-  if Size(R1) = Size(R2) and Columns(R1) = Columns(R2)
-      and Rows(R1) = Rows(R2) then
-    mat := Matrix(R1);
-    m := Length(mat[1]);
-    n := Length(mat);
-
-    if R1 = R2 then
-      g := UnderlyingSemigroup(R1);
-      return RMSIsoByTripleNC(R1, R2, [(),
-                                       IdentityMapping(g),
-                                       List([1 .. m + n], x -> One(g))]);
-    else
-      g1 := UnderlyingSemigroup(R1);
-      g2 := UnderlyingSemigroup(R2);
-      iso := IsomorphismGroups(g1, g2);
-
-      # for RMS without 0 the graphs are always isomorphic,
-      # being complete bipartite.
-
-      if not iso = fail then
-        isograph := DirectProduct(SymmetricGroup(m), SymmetricGroup(n));
-        #all isomorphisms from g1 to g2
-        isogroup := List(Elements(AutomorphismGroup(g1)), x -> x * iso);
-
-        #find an induced function, if there is one
-        RMSInducedFunction := SEMIGROUPS.RMSInducedFunction;
-        for l in isograph do
-          for g in isogroup do
-            for tup in Elements(g2) do
-              map := RMSInducedFunction(R1, R2, l, g, tup);
-              if map <> fail then
-                return RMSIsoByTripleNC(R1, R2, [l, g, map]);
-              fi;
-            od;
-          od;
-        od;
-      fi;
+InstallMethod(IsomorphismSemigroups,
+"for finite whole family Rees matrix semigroups",
+[IsReesMatrixSemigroup and IsWholeFamily and IsFinite,
+ IsReesMatrixSemigroup and IsWholeFamily and IsFinite],
+function(S, T)
+  local G, H, isoG, invG, s, first, isoH, invH, t, second, iso, mat, m, n, f,
+  isograph, isogroup, RMSInducedFunction, map, l, g, tup;
+
+  G := UnderlyingSemigroup(S);
+  H := UnderlyingSemigroup(T);
+
```

Comment:
Seeing as we might be able to return `fail` in the next few lines, even if it's not over a perm group, I've put this perm group check a few lines lower down.